### PR TITLE
feat: consider weight in score calculation

### PR DESCRIPTION
**Pull request by mariajgrimaldi**
### Description
This PR changes what is considered the final score for the student. Before these changes, we considered the `raw score` as the final score for the student without considering the weight configured. Now, we calculate the final score based on the weight configured in studio. This change has a downside: after calculating the new score, we could get numbers like 5.3 which cannot be saved in the submissions API so we decided to round to the nearest integer. 

### How to test
0. Configure grading in your course as follows:
image
1. Install this branch and add the xblock to your course
2. Configure your mindmap with your preferred weight. For this test, we're using 10 (the default):
image
3. As a student, submit a mindmap
image
4. As course staff, go to the component in the LMS and grade the submission:
image
5. As a student, check out your grade:
image
From the progress page:
image
Since the problem weight is 100, then our grade is 50. If the problem weight is 50, then our grade would be 25% and so on. Now, the subsection grade is our weighted score.

### CODE REVIEW

File: mindmap/mindmap.py [hunk 1]
```diff
@@ -634,7 +655,7 @@ def validate_score(points: int, weight: int) -> None:
 
         if weight:
             try:
-                weight = float(weight)
+                weight = int(weight)
```
Comment (Ian2012) [inline mindmap/mindmap.py:658]: shouldn't this be a float?

File: mindmap/public/js/src/mindmap.js [hunk 2]
```diff
@@ -84,7 +85,8 @@ function MindMapXBlock(runtime, element, context) {
                 gettext("Username"),
                 gettext("Uploaded"),
                 gettext("Submission Status"),
-                gettext("Grade"),
+                gettext("Raw score"),
+                gettext("Weighted score"),
```
Comment (johnvente) [inline mindmap/public/js/src/mindmap.js:89]: I think these two columns needs translations and update the translations lines

File: mindmap/mindmap.py (hunk 1 above)
Comment (mariajgrimaldi) [reply mindmap/mindmap.py:658]: I don't think so. The submissions API doesn't allow decimal values for scores, so when calculating the weighted score we should avoid decimal values. Unless you think of a use case we're missing when using integers.

**Comment by BryanttV** [comment]
LGTM!

**Review by BryanttV — APPROVED** [review]

File: mindmap/public/js/src/mindmap.js (hunk 2 above)
Comment (mariajgrimaldi) [reply mindmap/public/js/src/mindmap.js:89]: Done!

**Comment by mariajgrimaldi** [comment]
I'll be merging this since all comments have been addressed!

**Pull request merged**